core/unit_test/TestGraph.hpp

```diff
@@ -498,7 +498,7 @@ TEST_F(TEST_CATEGORY_FIXTURE(graph), force_global_launch) {
   ASSERT_TRUE(contains(ex, data, functor_t::count));
 
   ASSERT_TRUE(validate_event_set(
-      [&]() { graph.~optional(); },
```

Comment:
This caused
```
In member function 'constexpr void std::_Optional_payload_base<_Tp>::_M_reset() [with _Tp = Kokkos::Experimental::Graph<Kokkos::Cuda>]',
    inlined from 'std::_Optional_payload<_Tp, false, _Copy, _Move>::~_Optional_payload() [with _Tp = Kokkos::Experimental::Graph<Kokkos::Cuda>; bool _Copy = false; bool _Move = false]' at /usr/include/c++/13/optional:441:38,
    inlined from 'std::_Optional_base<Kokkos::Experimental::Graph<Kokkos::Cuda>, false, false>::~_Optional_base()' at /usr/include/c++/13/optional:512:8,
    inlined from 'std::optional<Kokkos::Experimental::Graph<Kokkos::Cuda> >::~optional()' at /usr/include/c++/13/optional:707:7,
    inlined from 'virtual void Test::cuda_graph_force_global_launch_Test::TestBody()' at /builds/kokkos/kokkos/core/unit_test/TestGraph.hpp:512:1:
/usr/include/c++/13/optional:317:18: error: '*(unsigned char*)((char*)&graph + offsetof(std::optional<Kokkos::Experimental::Graph<Kokkos::Cuda> >,std::optional<Kokkos::Experimental::Graph<Kokkos::Cuda> >::<unnamed>.std::_Optional_base<Kokkos::Experimental::Graph<Kokkos::Cuda>, false, false>::_M_payload.std::_Optional_payload<Kokkos::Experimental::Graph<Kokkos::Cuda>, false, false, false>::<unnamed>.std::_Optional_payload<Kokkos::Experimental::Graph<Kokkos::Cuda>, true, false, false>::<unnamed>.std::_Optional_payload_base<Kokkos::Experimental::Graph<Kokkos::Cuda> >::_M_engaged))' may be used uninitialized [-Werror=maybe-uninitialized]
  317 |         if (this->_M_engaged)
      |            ~~~~~~^~~~~~~~~~
/builds/kokkos/kokkos/core/unit_test/TestGraph.hpp: In member function 'virtual void Test::cuda_graph_force_global_launch_Test::TestBody()':
/builds/kokkos/kokkos/core/unit_test/TestGraph.hpp:458:62: note: '*(unsigned char*)((char*)&graph + offsetof(std::optional<Kokkos::Experimental::Graph<Kokkos::Cuda> >,std::optional<Kokkos::Experimental::Graph<Kokkos::Cuda> >::<unnamed>.std::_Optional_base<Kokkos::Experimental::Graph<Kokkos::Cuda>, false, false>::_M_payload.std::_Optional_payload<Kokkos::Experimental::Graph<Kokkos::Cuda>, false, false, false>::<unnamed>.std::_Optional_payload<Kokkos::Experimental::Graph<Kokkos::Cuda>, true, false, false>::<unnamed>.std::_Optional_payload_base<Kokkos::Experimental::Graph<Kokkos::Cuda> >::_M_engaged))' was declared here
  458 |   std::optional<Kokkos::Experimental::Graph<TEST_EXECSPACE>> graph;
      |                                                              ^~~~~
In destructor 'std::__shared_count<_Lp>::~__shared_count() [with __gnu_cxx::_Lock_policy _Lp = __gnu_cxx::_S_atomic]',
    inlined from 'std::__shared_ptr<_Tp, _Lp>::~__shared_ptr() [with _Tp = Kokkos::Impl::GraphImpl<Kokkos::Cuda>; __gnu_cxx::_Lock_policy _Lp = __gnu_cxx::_S_atomic]' at /usr/include/c++/13/bits/shared_ptr_base.h:1524:1,
    inlined from 'std::shared_ptr<Kokkos::Impl::GraphImpl<Kokkos::Cuda> >::~shared_ptr()' at /usr/include/c++/13/bits/shared_ptr.h:175:7,
    inlined from 'Kokkos::Experimental::Graph<Kokkos::Cuda>::~Graph()' at /builds/kokkos/kokkos/core/src/Kokkos_Graph.hpp:43:22,
    inlined from 'constexpr void std::_Optional_payload_base<_Tp>::_M_destroy() [with _Tp = Kokkos::Experimental::Graph<Kokkos::Cuda>]' at /usr/include/c++/13/optional:287:38,
    inlined from 'constexpr void std::_Optional_payload_base<_Tp>::_M_reset() [with _Tp = Kokkos::Experimental::Graph<Kokkos::Cuda>]' at /usr/include/c++/13/optional:318:11,
    inlined from 'constexpr void std::_Optional_payload_base<_Tp>::_M_reset() [with _Tp = Kokkos::Experimental::Graph<Kokkos::Cuda>]' at /usr/include/c++/13/optional:315:16,
    inlined from 'std::_Optional_payload<_Tp, false, _Copy, _Move>::~_Optional_payload() [with _Tp = Kokkos::Experimental::Graph<Kokkos::Cuda>; bool _Copy = false; bool _Move = false]' at /usr/include/c++/13/optional:441:38,
    inlined from 'std::_Optional_base<Kokkos::Experimental::Graph<Kokkos::Cuda>, false, false>::~_Optional_base()' at /usr/include/c++/13/optional:512:8,
    inlined from 'std::optional<Kokkos::Experimental::Graph<Kokkos::Cuda> >::~optional()' at /usr/include/c++/13/optional:707:7,
    inlined from 'virtual void Test::cuda_graph_force_global_launch_Test::TestBody()' at /builds/kokkos/kokkos/core/unit_test/TestGraph.hpp:512:1:
/usr/include/c++/13/bits/shared_ptr_base.h:1070:8: error: '((std::__shared_count<__gnu_cxx::_S_atomic>*)((char*)&graph + offsetof(std::optional<Kokkos::Experimental::Graph<Kokkos::Cuda> >,std::optional<Kokkos::Experimental::Graph<Kokkos::Cuda> >::<unnamed>.std::_Optional_base<Kokkos::Experimental::Graph<Kokkos::Cuda>, false, false>::_M_payload.std::_Optional_payload<Kokkos::Experimental::Graph<Kokkos::Cuda>, false, false, false>::<unnamed>.std::_Optional_payload<Kokkos::Experimental::Graph<Kokkos::Cuda>, true, false, false>::<unnamed>.std::_Optional_payload_base<Kokkos::Experimental::Graph<Kokkos::Cuda> >::_M_payload)))[1].std::__shared_count<>::_M_pi' may be used uninitialized [-Werror=maybe-uninitialized]
 1070 |         if (_M_pi != nullptr)
      | ^       ~
/builds/kokkos/kokkos/core/unit_test/TestGraph.hpp: In member function 'virtual void Test::cuda_graph_force_global_launch_Test::TestBody()':
/builds/kokkos/kokkos/core/unit_test/TestGraph.hpp:458:62: note: '((std::__shared_count<__gnu_cxx::_S_atomic>*)((char*)&graph + offsetof(std::optional<Kokkos::Experimental::Graph<Kokkos::Cuda> >,std::optional<Kokkos::Experimental::Graph<Kokkos::Cuda> >::<unnamed>.std::_Optional_base<Kokkos::Experimental::Graph<Kokkos::Cuda>, false, false>::_M_payload.std::_Optional_payload<Kokkos::Experimental::Graph<Kokkos::Cuda>, false, false, false>::<unnamed>.std::_Optional_payload<Kokkos::Experimental::Graph<Kokkos::Cuda>, true, false, false>::<unnamed>.std::_Optional_payload_base<Kokkos::Experimental::Graph<Kokkos::Cuda> >::_M_payload)))[1].std::__shared_count<>::_M_pi' was declared here
  458 |   std::optional<Kokkos::Experimental::Graph<TEST_EXECSPACE>> graph;
      |                              
```

Reply:
This is not legal.  Do `reset()` instead.